### ISSUE TITLE
Encoding problem

### ISSUE THREAD

**Issue by joco73** [arashnorouzi/Moon-APNS#5]
Hi,

Great package.

When we have languages that use f.ex. åäö we can't have ascii encoding, we then need UTF8, on your blog a comment has a solution that works for us (see below), but would be great to have this in the package - as a config. parameter or something else.

On line 296 in ‘PushNotification.cs’ file, I have modified your code as follows..

// String length
byte[] apnMessageLength = BitConverter.GetBytes((Int16)Encoding.UTF8.GetBytes(apnMessage).Length);

and on line 305 was modified as

// Write the message
memoryStream.Write(Encoding.UTF8.GetBytes(apnMessage), 0, Encoding.UTF8.GetBytes(apnMessage).Length);

/Jonathan

**Comment by soundtina** [comment]
@joco73  yeah it's helpful. Now i can send Chinese to my phone. That's cool.
